### PR TITLE
bpo-21360: mailbox.Maildir now ignores files with a leading dot

### PR DESCRIPTION
**Pull request by ZackerySpytz**
The maildir format specification states that files with a leading dot
should be ignored.

https://bugs.python.org/issue21360

### CODE REVIEW

**Review by remilapeyre — APPROVED** [review]
Thanks!

File: Lib/test/test_mailbox.py [hunk 1]
```diff
@@ -672,6 +672,15 @@ def test_initialize_existing(self):
         self._box = mailbox.Maildir(self._path)
         self._check_basics()
 
+    def test_filename_leading_dot(self):
```
Comment (serhiy-storchaka) [inline Lib/test/test_mailbox.py:684]: This test is specific to `Maildir`. Please move it to `TestMaildir`.

File: Lib/test/test_mailbox.py [hunk 2]
```diff
@@ -672,6 +672,15 @@ def test_initialize_existing(self):
...
+        self.tearDown()
+        for subdir in '', 'tmp', 'new', 'cur':
+            os.mkdir(os.path.normpath(os.path.join(self._path, subdir)))
+        with open(os.path.join(self._path, 'new', '.foo'), 'w') as f:
```
Comment (serhiy-storchaka) [inline Lib/test/test_mailbox.py]: Maybe test other subdirs too?

**Comment by remilapeyre** [comment]
Hi @ZackerySpytz, I think this PR got lost but there is just a few things to address before it can be merged.

**Review by serhiy-storchaka — APPROVED** [review]

**Pull request merged**